src/runner.ts

```diff
@@ -102,155 +104,161 @@ export interface IRunnerOptions {
     version?: boolean;
 }
 
-export class Runner {
-    private static trimSingleQuotes(str: string) {
-        return str.replace(/^'|'$/g, "");
-    }
-
-    constructor(private options: IRunnerOptions, private outputStream: NodeJS.WritableStream) { }
+export const enum Status {
+    Ok = 0,
+    FatalError = 1,
+    LintError = 2,
+}
 
-    public run(onComplete: (status: number) => void) {
-        if (this.options.version) {
-            this.outputStream.write(Linter.VERSION + "\n");
-            return onComplete(0);
+export async function run(options: Options, outputStream: NodeJS.WritableStream, errorStream: NodeJS.WritableStream): Promise<Status> {
```

Comment:
According to https://github.com/palantir/tslint/issues/2446#issuecomment-290613113, anything not in `index` is not public.